Running DenseNet model on covid-chestxray dataset

oplatek opened the pull request:
See the second commit and README.md for description

oplatek committed:
add gitignore from PL

oplatek committed:
running DenseNet model on covid-chestxray dataset

oplatek commented:
The readme is more readable here https://github.com/oplatek/lightning-Covid19/blob/feature/baseline-model/README.md

oplatek commented:
Inspired by https://github.com/ieee8023/covid-chestxray-dataset/issues/15#issuecomment-599263502

Borda committed:
Merge branch 'master' into feature/baseline-model

pep8speaks commented:
Hello @oplatek! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`lightning_covid19/model/base_densenet.py`](https://github.com/PyTorchLightning/lightning-Covid19/blob/4c54dfc595dae4e9d177e3c7fc26249ee94cd21f/lightning_covid19/model/base_densenet.py):

> [Line 3:80](https://github.com/PyTorchLightning/lightning-Covid19/blob/4c54dfc595dae4e9d177e3c7fc26249ee94cd21f/lightning_covid19/model/base_densenet.py#L3): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 36:80](https://github.com/PyTorchLightning/lightning-Covid19/blob/4c54dfc595dae4e9d177e3c7fc26249ee94cd21f/lightning_covid19/model/base_densenet.py#L36): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (92 > 79 characters)
> [Line 37:80](https://github.com/PyTorchLightning/lightning-Covid19/blob/4c54dfc595dae4e9d177e3c7fc26249ee94cd21f/lightning_covid19/model/base_densenet.py#L37): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (92 > 79 characters)
> [Line 59:80](https://github.com/PyTorchLightning/lightning-Covid19/blob/4c54dfc595dae4e9d177e3c7fc26249ee94cd21f/lightning_covid19/model/base_densenet.py#L59): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (98 > 79 characters)
> [Line 86:80](https://github.com/PyTorchLightning/lightning-Covid19/blob/4c54dfc595dae4e9d177e3c7fc26249ee94cd21f/lightning_covid19/model/base_densenet.py#L86): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (98 > 79 characters)
> [Line 105:80](https://github.com/PyTorchLightning/lightning-Covid19/blob/4c54dfc595dae4e9d177e3c7fc26249ee94cd21f/lightning_covid19/model/base_densenet.py#L105): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (88 > 79 characters)
> [Line 129:80](https://github.com/PyTorchLightning/lightning-Covid19/blob/4c54dfc595dae4e9d177e3c7fc26249ee94cd21f/lightning_covid19/model/base_densenet.py#L129): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (91 > 79 characters)
> [Line 133:80](https://github.com/PyTorchLightning/lightning-Covid19/blob/4c54dfc595dae4e9d177e3c7fc26249ee94cd21f/lightning_covid19/model/base_densenet.py#L133): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 142:80](https://github.com/PyTorchLightning/lightning-Covid19/blob/4c54dfc595dae4e9d177e3c7fc26249ee94cd21f/lightning_covid19/model/base_densenet.py#L142): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (83 > 79 characters)
> [Line 143:80](https://github.com/PyTorchLightning/lightning-Covid19/blob/4c54dfc595dae4e9d177e3c7fc26249ee94cd21f/lightning_covid19/model/base_densenet.py#L143): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (91 > 79 characters)
> [Line 151:80](https://github.com/PyTorchLightning/lightning-Covid19/blob/4c54dfc595dae4e9d177e3c7fc26249ee94cd21f/lightning_covid19/model/base_densenet.py#L151): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (104 > 79 characters)
> [Line 157:80](https://github.com/PyTorchLightning/lightning-Covid19/blob/4c54dfc595dae4e9d177e3c7fc26249ee94cd21f/lightning_covid19/model/base_densenet.py#L157): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (117 > 79 characters)
> [Line 165:80](https://github.com/PyTorchLightning/lightning-Covid19/blob/4c54dfc595dae4e9d177e3c7fc26249ee94cd21f/lightning_covid19/model/base_densenet.py#L165): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (103 > 79 characters)
> [Line 167:80](https://github.com/PyTorchLightning/lightning-Covid19/blob/4c54dfc595dae4e9d177e3c7fc26249ee94cd21f/lightning_covid19/model/base_densenet.py#L167): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (96 > 79 characters)
> [Line 168:80](https://github.com/PyTorchLightning/lightning-Covid19/blob/4c54dfc595dae4e9d177e3c7fc26249ee94cd21f/lightning_covid19/model/base_densenet.py#L168): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (102 > 79 characters)
> [Line 170:80](https://github.com/PyTorchLightning/lightning-Covid19/blob/4c54dfc595dae4e9d177e3c7fc26249ee94cd21f/lightning_covid19/model/base_densenet.py#L170): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (89 > 79 characters)
> [Line 175:80](https://github.com/PyTorchLightning/lightning-Covid19/blob/4c54dfc595dae4e9d177e3c7fc26249ee94cd21f/lightning_covid19/model/base_densenet.py#L175): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (83 > 79 characters)
> [Line 178:16](https://github.com/PyTorchLightning/lightning-Covid19/blob/4c54dfc595dae4e9d177e3c7fc26249ee94cd21f/lightning_covid19/model/base_densenet.py#L178): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 188:80](https://github.com/PyTorchLightning/lightning-Covid19/blob/4c54dfc595dae4e9d177e3c7fc26249ee94cd21f/lightning_covid19/model/base_densenet.py#L188): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (91 > 79 characters)
> [Line 191:80](https://github.com/PyTorchLightning/lightning-Covid19/blob/4c54dfc595dae4e9d177e3c7fc26249ee94cd21f/lightning_covid19/model/base_densenet.py#L191): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (90 > 79 characters)
> [Line 194:80](https://github.com/PyTorchLightning/lightning-Covid19/blob/4c54dfc595dae4e9d177e3c7fc26249ee94cd21f/lightning_covid19/model/base_densenet.py#L194): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (91 > 79 characters)

* In the file [`lightning_covid19/utils.py`](https://github.com/PyTorchLightning/lightning-Covid19/blob/4c54dfc595dae4e9d177e3c7fc26249ee94cd21f/lightning_covid19/utils.py):

> [Line 14:80](https://github.com/PyTorchLightning/lightning-Covid19/blob/4c54dfc595dae4e9d177e3c7fc26249ee94cd21f/lightning_covid19/utils.py#L14): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (82 > 79 characters)
> [Line 15:17](https://github.com/PyTorchLightning/lightning-Covid19/blob/4c54dfc595dae4e9d177e3c7fc26249ee94cd21f/lightning_covid19/utils.py#L15): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 17:80](https://github.com/PyTorchLightning/lightning-Covid19/blob/4c54dfc595dae4e9d177e3c7fc26249ee94cd21f/lightning_covid19/utils.py#L17): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (88 > 79 characters)

* In the file [`run_baseline_densenet.py`](https://github.com/PyTorchLightning/lightning-Covid19/blob/4c54dfc595dae4e9d177e3c7fc26249ee94cd21f/run_baseline_densenet.py):

> [Line 21:80](https://github.com/PyTorchLightning/lightning-Covid19/blob/4c54dfc595dae4e9d177e3c7fc26249ee94cd21f/run_baseline_densenet.py#L21): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 22:4](https://github.com/PyTorchLightning/lightning-Covid19/blob/4c54dfc595dae4e9d177e3c7fc26249ee94cd21f/run_baseline_densenet.py#L22): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 22:80](https://github.com/PyTorchLightning/lightning-Covid19/blob/4c54dfc595dae4e9d177e3c7fc26249ee94cd21f/run_baseline_densenet.py#L22): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (97 > 79 characters)

Borda commented on data/.gitignore:
```diff
@@ -0,0 +1 @@
+*
```
```suggestion
```

oplatek replied:
It is the directory where the data are downloaded - I do not want to see any of it.

@Borda  Why do you want to remove `*`?

Borda replied:
Directory with data is fine, but here is just a * meaning anything, right?

oplatek replied:
Right - `*` means anything - the semantics here -  under the `data` directory you can paste/download anything and it will be ignored.
Note that the `.gitignore` is in the data subdirectory.

I can change it to the default data_root under the data directory for covid19 if it bothers you 
i.e. `covid-chestxray-dataset`

Borda replied:
ohh sorry, I missed that it not in the root
so then rather place this folder in the root ignore

oplatek replied:
@Borda ok I will add .gitkeep and move the ignore statement to the global gitignore

oplatek replied:
done

Borda commented on lightning_covid19/model/base_densenet.py:
```diff
@@ -0,0 +1,194 @@
+"""
```
just densenet.py :]

oplatek replied:
done

Borda commented on lightning_covid19/model/base_densenet.py:
```diff
@@ -0,0 +1,194 @@
+import torchxrayvision as xrv
+from pytorch_lightning.core import LightningModule
+
+from ..utils import run
```
```suggestion
from lightning_covid19.utils import run
```

oplatek replied:
done

Borda commented on run_baseline_densenet.py:
```diff
@@ -0,0 +1,34 @@
+                  fast_dev_run=hparams.fast_dev_run,
+                  )
+trainer.fit(denseNet_model)
+trainer.test()
```
lets move all this under `if __name__ == '__main__':`

oplatek replied:
done

Borda reviewed (changes requested):
these proposals should help tests pass

shpotes commented on lightning_covid19/model/base_densenet.py:
```diff
@@ -0,0 +1,194 @@
+
+        # split the dataset
+        self.train_set, self.val_set, self.test_set = \
+            torch.utils.data.random_split(covid19, [n_train, n_valid, n_test])
```
Possible data leakage?
On the original dataset, there are several images from the same patient [see for example patient number 2](https://github.com/ieee8023/covid-chestxray-dataset/blob/master/metadata.csv#L2-L5) 
Should we take this into account when splitting the data?

oplatek replied:
Great catch! We definitely should..... 

The scenario with covid19 is that a completely new person is diagnosed.
It is hardly acceptable to assume that our model has seen the xray image of that new person before.

Let me try to find a solution quickly - otherwise, I need to hand it over to somebody I am bus today.

oplatek replied:
Unfortunately not a simple fix without changing torchxrayvision https://github.com/mlmed/torchxrayvision/issues/2

I opened https://github.com/PyTorchLightning/lightning-Covid19/issues/11 so we won't forget on that - I suggest to keep it as a bug and accept this MR.

@Borda thoughts?

oplatek committed:
code style changes

rename base_densenet.py -> densenet.py
pep fixes
ifmain in run_baseline_densenet.py

oplatek committed:
ignore venv and PL logs

oplatek commented:
There is an example dataloader @Borda should we close #2 ?

oplatek committed:
flake8 fixes

oplatek committed:
use only global gitignore

Borda commented on lightning_covid19/utils.py:
```diff
@@ -0,0 +1,25 @@
+import numpy as np
+
+
+def run(cmd, stderr=subprocess.STDOUT):
```
what is this `run` for?

Borda reviewed (approved):
Just move the experiment to Wiki

Borda committed:
Update README.md
